packages/flutter/lib/src/semantics/semantics_event.dart

```diff
@@ -159,3 +159,69 @@ class TapSemanticEvent extends SemanticsEvent {
   @override
   Map<String, dynamic> getDataMap() => const <String, dynamic>{};
 }
+
+/// An event to move the accessibility focus.
+///
+/// Using this API is generally not recommended, as it may break the consistency of the
+/// accessibiliy focus of the platform.
+///
+/// Some possibile use cases:
+/// For example, the currently focused rendering object is replaced by another rendering
+/// object. In general, such design should be avoided if possible. If not, one may want
+/// to refocus the newly added rendering object.
+///
+/// Some examples that are not recommended:
+/// When a new popup or dropdown opens, moving the focus in these cases may confuse users
+/// and make it less accessible.
+///
+/// {@tool snippet}
+///
+/// The following code snippet shows how one can request focus on a
+/// certain widget.
+/// ```dart
+/// class MyWidget extends StatefulWidget {
+///   const MyWidget({super.key});
+///
+///   @override
+///   State<MyWidget> createState() => _MyWidgetState();
+/// }
+///
+/// class _MyWidgetState extends State<MyWidget> {
+///   bool noticeAccepted = false;
+///   final GlobalKey mykey= GlobalKey();
+///
+///   @override
+///   void initState() {
+///     super.initState();
+///     WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+///       mykey.currentContext?.findRenderObject()?.sendSemanticsEvent(const FocusSemanticEvent());
```

Comment:
Questions I have here that probably should be answered by the doc: What if the renderObject doesn't own a semantics node? Where does focus go?

Reply:
sendSemanticsEvent will look up for the first parent render object with a non-null semantic node.

And it doesn't work for the textField, so I am thinking about overriding textField 's `sendSemanticsEvent` as a workaround, do you think it's acceptable?

And do yo know some other cases like the textField?